sail/g6/Player.java

```diff
@@ -38,66 +36,59 @@ public Point chooseStartingLocation(Point wind_direction, Long seed, int t) {
     public void init(List<Point> group_locations, List<Point> targets, int id) {
         this.targets = targets;
         this.id = id;
+        isVisited = new int[targets.size()][group_locations.size()];
     }
 
     @Override
     public Point move(List<Point> group_locations, int id, double dt, long time_remaining_ms) {
 
-        // Check if we've reached the target.
         if (this.path.size() != 0 && visited_set.get(id).contains(nextTarget)) {
             prevLoc.add(0, path.get(0));
             path.remove(0);
         }
         else if (this.path.size() != 0) {
-            // System.out.println("Moving to set target.");
-
-            // Check if we crossed the intermediate point.
-            // This won't be a target point because that is already checked for in the
-            // first "if".
             if (checkCrossedInterpolation(path.get(0), prevLoc.get(0), group_locations.get(id))) {
                 prevLoc.add(0, path.get(0));
                 path.remove(0);
-                // We are guaranteed that there is another point in "path".
-            }
-
-            if (path.size() == 0) {
-                // We should never come here.
-                // If we do, then it means that the first "if" that checks if
-                // we crossed a target is buggy.
-                System.out.println("Whaaaaaa");
             }
-
-            // Continue moving to the intermediate point or move to the target point.
             return moveInPath(group_locations.get(id), path.get(0));
         }
 
         if(visited_set != null && visited_set.get(id).size() == targets.size()) {
-            // This is if we have finished visiting all targets.
             nextTarget = targets.size();
             return findPathAndMove(group_locations.get(id), initial);
         }
         else {
-            // System.out.println("Selecting next target.");
-
-            // Here is the logic to decide which target to head to next.
-            // Pick the closest target.
+            for (int i = 0; i < targets.size(); i++) {
+                for (int j = 0; j < group_locations.size(); j++) {
+                    if (isVisited[i][j] == 1) continue;
+                    if (Point.getDistance(targets.get(i), group_locations.get(j)) < 0.1) {

```

Comment:
This should be 0.01, right?
This computation may not be accurate since the player could have overshot their target.

Reply:
Because I think 0.01 is too small, I decide to use 0.1 in case that some players have already passed the target.

Reply:
But that means that players may not have crossed the target but we still consider that they do?

Reply:
Yes but I think it's not a big issue because 0.1 is not a quite large number.